File: hw/fuentes/U2/HW13ObjectCRUD/VinylRecordSale/src/main/java/ec/edu/espe/model/VinylRecord.java

package ec.edu.espe.model;

/**
 *
 * @author jorge
 */
public class VinylRecord {
    private String title;
    private String artist;
    private int year;
    private String genre;
    private String duration;
    private String label;
    private String country;
    private String color;
    private float basePrice;
    private boolean limitedEdition;
    private float priceAdjustment;
    private float finalPrice;

    public VinylRecord(String title, String artist, int year, String genre, String duration, String label, String country, String color, float basePrice, boolean limitedEdition) {
        this.title = title;
        this.artist = artist;
        this.year = year;
        this.genre = genre;
        this.duration = duration;
        this.label = label;
        this.country = country;
        this.color = color;
        this.basePrice = basePrice;
        this.limitedEdition = limitedEdition;
        this.priceAdjustment = calculatePriceAdjustment();
        this.finalPrice = basePrice + this.priceAdjustment;
    }
    
    private float calculatePriceAdjustment() {
        float adjustment = 0.0f;
        if (limitedEdition) {
            adjustment += basePrice * 0.5f;
        }
        if (!color.equals("black")) {
            adjustment += 10.0f;
        }
        return adjustment;
    }

    public String getTitle() {
        return title;
    }

    public String getArtist() {
        return artist;
    }

    public int getYear() {
        return year;
    }

    public String getGenre() {
        return genre;
    }

    public String getDuration() {
        return duration;
    }

    public String getLabel() {
        return label;
    }

    public String getCountry() {
        return country;
    }

    public String getColor() {
        return color;
    }

    public float getBasePrice() {
        return basePrice;
    }

    public boolean isLimitedEdition() {
        return limitedEdition;
    }

    public float getPriceAdjustment() {
        return priceAdjustment;
    }

    public float getFinalPrice() {
        return finalPrice;
    }
    
    
    
}

File: hw/fuentes/U2/HW13ObjectCRUD/VinylRecordSalePY/VinylRecord.py
class VinylRecord:
    def __init__(self, title, artist, year, genre, duration, label, country, color, base_price, limited_edition):
        self.title = title
        self.artist = artist
        self.year = year
        self.genre = genre
        self.duration = duration
        self.label = label
        self.country = country
        self.color = color
        self.base_price = base_price
        self.limited_edition = limited_edition
        self.price_adjustment = self.calculate_price_adjustment()
        self.final_price = self.base_price + self.price_adjustment

    def calculate_price_adjustment(self):
        adjustment = 0.0
        if self.limited_edition:
            adjustment += self.base_price * 0.5
        if self.color.lower() != 'black':
            adjustment += 10.0
        return adjustment

    def to_dict(self):
        return {
            "title": self.title,
            "artist": self.artist,
            "year": self.year,
            "genre": self.genre,
            "duration": self.duration,
            "label": self.label,
            "country": self.country,
            "color": self.color,
            "basePrice": self.base_price,
            "limitedEdition": self.limited_edition,
            "priceAdjustment": self.price_adjustment,
            "finalPrice": self.final_price
        }
